### PR TITLE
Beatmap import

### PR DESCRIPTION
**Pull request by ddevault**
Note that this is based on #44, new commits start at dfaca3b.

### CODE REVIEW

**Comment by SirCmpwn** [comment]
TODO: ArchiveReaders should implement IDisposable

File: osu.Game/Beatmaps/IO/OszArchiveReader.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        }
+    
+        private ZipFile Archive { get; set; }
+        private string[] Beatmaps { get; set; }
+        private Beatmap FirstMap { get; set; }
+    
+        public OszArchiveReader(Stream archive)
+        {
+            Archive = ZipFile.Read(archive);
+            Beatmaps = Archive.Entries.Where(e => e.FileName.EndsWith(".osu"))
```
Comment (peppy) [inline osu.Game/Beatmaps/IO/OszArchiveReader.cs]: loading of beatmaps should be done lazily on first usage imo. also should be Beatmap[] not string[].

Comment (SirCmpwn) [inline osu.Game/Beatmaps/IO/OszArchiveReader.cs]: I didn't want to necessarily have to *parse* beatmaps at this stage, just get a list of their files.

**Comment by SirCmpwn** [comment]
TODO: reading from zips like this

**Pull request closed without merging**